source/particle/world.cc

```diff
@@ -773,13 +712,12 @@ namespace aspect
     void
     World<dim>::initialize_particles()
     {
-#if !DEAL_II_VERSION_GTE(9,4,0)
+!DEAL_II_VERSION_GTE(9,4,0)
```

Comment:
You accidentally removed the `#if` here but left the condition in place. But this is not right in any case. The condition here is "if the deal.II version is NOT >=9.4", i.e., if the deal.II version is older than the minimum supported version. You can remove the whole block between `#if` and `#endif`.